### DISCUSSION TITLE
Aligning text with widgets

### DISCUSSION THREAD

**Discussion by avalon60** [TomSchimansky/CustomTkinter#1381] [Q&A]
Hi. This is an area which has caused some frustration. Below is an image which hopefully demonstrates the problem.
For the labels I am using sticky='e' and for the entry/select widgets I use sticky='w' 

Code snippet:

```

        default_padx = 0
        default_pady = 10
        row = 0
...
        column = 0

        lbl_mod_database_type = ctk.CTkLabel(master=self.frm_mod_conn_type, text='Database Type:')
        lbl_mod_database_type.grid(row=row, column=column, padx=default_padx, pady=default_pady, sticky='e')
        column += 1

        self.opm_mod_database_type = ctk.CTkOptionMenu(master=self.frm_mod_conn_type,
                                                       values=self.mvc_controller.database_type_descriptors())
        self.opm_mod_database_type.grid(row=row, column=column, padx=default_padx, pady=default_pady, sticky='sw')
...
        row += 1
        column = 0
        lbl_mod_client_tool = ctk.CTkLabel(master=self.frm_mod_conn_type, text='Client Tool:')
        lbl_mod_client_tool.grid(row=row, column=column, padx=default_padx, pady=default_pady, sticky='e')

        column += 1
        self.opm_mod_client_tool = ctk.CTkOptionMenu(master=self.frm_mod_conn_type,
                                                     command=self.mvc_controller.set_client_options_state,
                                                     values=self.mvc_controller.client_tools_name_list)
        self.opm_mod_client_tool.grid(row=row, column=column, padx=default_padx, pady=default_pady, sticky='w')

```
I kind of expected the "sticky" would align the right end of the label text, but it doesn't seem to do this. I know I could use padx, but this seems a bit imprecise (squint of the eye process). Am I missing something here?

Thanks in advance.

![image](https://user-images.githubusercontent.com/89534395/227719171-31d8a0a3-7919-49e7-843e-a6c9fbc0e42b.png)

**Comment by Akascape** [comment]
@avalon60 We have to use rowconfigure and columnconfigure properly for the frame.

**Reply by avalon60** [reply]
Hi Akash - I'm not sure how that helps in this context? I thought that they just controlled how the regions react to resizing, by the use of weight settings?

**Reply by Akascape** [reply]
@avalon60 Yes. Generally sticky only works if we specify the weights to the row/columns.

**Reply by avalon60** [reply]
I think using grid_columnconfigure will only help, where you place the items in adjacent frames. I just found the answer is to use the
anchor='e' parameter setting on the label, when I instantiate the CTkLabel:

`        lbl_mod_client_tool = ctk.CTkLabel(master=self.frm_mod_conn_type, text='Client Tool:', anchor='e')`

**Reply by avalon60** [reply]
It looks tidier now. 

It's a shame that there doesn't seem to be an option to place the text on switches to the left. It kind of jars the flow in my mind. I know I could use an empty string for the CTkSwitch label text, and add a CTKLabel, but that's not quite as convenient.

![image](https://user-images.githubusercontent.com/89534395/227728778-96170a2a-d1eb-4ee4-8266-aaac6be092f6.png)
